jetty-server/src/main/java/org/eclipse/jetty/server/Response.java

```diff
@@ -622,6 +595,48 @@ public void sendRedirect(String location) throws IOException
         sendRedirect(HttpServletResponse.SC_MOVED_TEMPORARILY, location);
     }
 
+    /**
+     * Common point to generate a proper "Location" header for redirects.
+     *
+     * @param request the request the redirect should be based on (needed when relative locations are provided, so that
+     * server name, scheme, port can be built out properly)
+     * @param location the location URL to redirect to (can be a relative path)
+     * @return the full redirect "Location" URL (including scheme, host, port, path, etc...)
+     */
+    public static String toRedirectURL(final HttpServletRequest request, String location)
```

Comment:
Rename to `toRedirectURI()`, as RFC and Jetty 12 use "URI", not "URL".

Drop `final` for the first parameter.

Also, should clarify whether the `location` is already encoded or not?
E.g. `location: newPath?p=/oldPath` should the `/` be already encoded?

Perhaps it's just enough to change the variable name to `encodedLocation` to make it clear, but callers need to know.